db/view/view_update_generator.cc

```diff
@@ -24,6 +25,64 @@ static inline void inject_failure(std::string_view operation) {
 
 namespace db::view {
 
+class view_update_generator::progress_tracker final : public sstables::read_monitor_generator {
+    class read_monitor final : public  sstables::read_monitor {
+        sstables::shared_sstable _sst;
+        uint64_t& _pending_work;
+    public:
+        virtual void on_read_started(const sstables::reader_position_tracker& tracker) override {}
```

Comment:
Can we use this to have more fine-grained tracking? Staging sstables can be quite big, so the progress counter will be frozen for hours possibly if we track on a per-sstable basis only.

Reply:
Found out we can make it lightweight by keeping inactive pending work in a separate counter, and only iterating over the read_monitor for SSTables being processed. With your series to reduce the batch size to 4, that will be very cheap.